### PR TITLE
add backup jobs for DB

### PR DESCRIPTION
**Pull request by v1peractual**
Add backup job to mysql server profile
Fix puppet lint issues
Update pdk to v3

### CODE REVIEW

File: manifests/REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Reference
```
Comment (billglick) [inline manifests/REFERENCE.md]: I think this file was likely accidentally created and should be deleted.

File: manifests/backup.pp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      filename       => 'mysql_all_databases.dump',
+    }
+  } else {
+    $databases.each|$db| {
+      profile_backup::client::add_cmd_job { "mysql_${db}":
+        backup_command => "${backup_cmd} ${backup_cmd_options} ${db}",
+        filename       => "mysql_${db}.dump",
+      }
+    }
+  }
```
Comment (billglick) [inline manifests/backup.pp]: Interesting. I hadn't considered backup jobs for each individual database. 
I like it.

File: manifests/backup.pp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#
+# @example 
+#   include profile_mysql_server::backup
+class profile_mysql_server::backup (
+  String            $backup_cmd,
+  String            $backup_cmd_options,
+  Array[String]     $databases,
+) {
+  if empty($databases) {
+    profile_backup::client::add_cmd_job { 'mysql_all_databases':
```
Comment (billglick) [inline manifests/backup.pp]: In some other NCSA modules we are naming the backup job with the NCSA Puppet profile that is providing the backup job. The idea being it was more clear what Puppet module is providing the backup job.

If that approach makes sense, then I'd suggest changing this:
```suggestion
    profile_backup::client::add_cmd_job { 'profile_mysql_server_all_databases':
```

And similarly the name in line 28 would be adjusted to append `profile_mysql_server`.

**Review by billglick — CHANGES_REQUESTED** [review]
This generally looks good to me.

I did have a couple suggestions and comments.

And I'd like @jakerundall to do a review before merging this.

File: manifests/REFERENCE.md (hunk 1 above)
Comment (v1peractual) [reply manifests/REFERENCE.md]: Deleted.

File: manifests/backup.pp (hunk 2 above)
Comment (v1peractual) [reply manifests/backup.pp]: Allows for easier restoring of a specific database without having to pull the whole dump.

File: manifests/backup.pp (hunk 3 above)
Comment (v1peractual) [reply manifests/backup.pp]: renamed.

**Review by billglick — APPROVED** [review]

**Comment by v1peractual** [comment]
Jake is out until the 3rd. Do you want to wait until then?

**Comment by billglick** [comment]
I forgot Jake was out for several days. We do not need to wait for Jake.

This seems like a fairly safe change:
1. It does logical things by default.
2. It won't even attempt to backup if system doesn't yet have backups configured.

**Pull request merged**